zipline/pipeline/factors/factor.py

```diff
@@ -1205,15 +1205,20 @@ def compute(self, today, assets, out, close):
     '''
     dtype = float64_dtype
 
-    def __getattr__(self, attribute_name):
+    def __getattribute__(self, name):
+        outputs = object.__getattribute__(self, 'outputs')
+        if outputs is not NotSpecified and name in outputs:
+            return RecarrayField(factor=self, attribute=name)
+        else:
+            return super(CustomFactor, self).__getattribute__(name)
+
+    def __getattr__(self, name):
         if self.outputs is NotSpecified:
-            return getattr(super(CustomFactor, self), attribute_name)
-        if attribute_name in self.outputs:
-            return RecarrayField(factor=self, attribute=attribute_name)
+            return getattr(super(CustomFactor, self), name)
         else:
             raise AttributeError(
                 'Instance of {factor} has no output called {attr!r}.'.format(
-                    factor=type(self).__name__, attr=attribute_name,
+                    factor=type(self).__name__, attr=name,
                 )
             )
```

Comment:
I kept `__getattr__` here to make it clearer when we should show this error message